getting-started/setup-multi-node-auth.md

```diff
@@ -464,6 +464,36 @@ And finally add the role to all of the data nodes with [`distributed_exec`][dist
 CALL distributed_exec($$ CREATE ROLE testrole LOGIN $$);
 ```
 
+---
+## Data node maintenance job on the access node
+
+It is highly recommended that the access node is configured to run a
+maintenance job that regularly "heals" any non-completed distributed
+transactions. A distributed transaction can remain in a non-completed
+state in case a data node reboots or experiences temporary issues. The
+access node keeps a log of distributed transactions so that nodes that
+haven't yet completed their part of the distributed transaction can
+later complete it at the access node's request. The maintenance job
+can be run as a user-defined action (custom job):
+
+
+```sql
+CREATE OR REPLACE PROCEDURE data_node_maintenance(job_id int, config jsonb)
+LANGUAGE SQL AS
+$$
+    SELECT _timescaledb_internal.remote_txn_heal_data_node(fs.oid)
```

Comment:
I haven't tested this, so I just want to be sure, this function can't be called inside a transaction block, and I know there's an implicit transaction in single sql statement, but I'm not sure if that would cause issues here or if we'd need to write this as a separate function call per fs.oid with a PLPGSQL loop or something? Anyway, just wanted to make sure we'd tested this with multiple data nodes so that we know there's no txn issues in that case.

Reply:
So, if you do `SELECT myfunction();` or `CALL myproc()`, those functions/procedures will execute as part of a single transaction (unless the procedure itself does something funky with transactions). In any case, if there is no explicit `BEGIN;` prior to calling this procedure, there is no transaction block and the procedure will be executed. I did test this with two data nodes, and both the procedure and job executes successfully covering two data nodes (although no actual healing was done).

Obviously, manually executing this query:

```
SELECT _timescaledb_internal.remote_txn_heal_data_node(fs.oid)
    FROM pg_foreign_server fs, pg_foreign_data_wrapper fdw
    WHERE fs.srvfdw = fdw.oid
    AND fdw.fdwname = 'timescaledb_fdw';
```
is a single transaction, but still involves multiple data nodes.

That said, this is @cevian's code and design, so I am not really sure myself what is correct here. A comment in the `remote_txn_heal_data_node` function gives me pause: 

```
	/*
	 * This function cannot be called inside a transaction block since effects
	 * cannot be rolled back
	 */
	PreventInTransactionBlock(true, "remote_txn_heal_data_node");

```
I assume that the intention here is to only execute this on one node per transaction as to ensure that there is not a situation where a failure on the success node means that changes on the first one cannot be rolled back (since this isn't executed under 2PC). I suspect that preventing a transaction block cannot protect against executing this on two data nodes in a single transaction.

Reply:
FWIW, should verify with @cevian, but I think `remote_txn_heal_data_node` should probably be turned into a procedure that starts a new transaction every time it is called. Otherwise, I think we need to keep some state in the function during the transaction that prevents calling this function multiple times in the same transaction.

Reply:
I think something like: 
```
CREATE OR REPLACE PROCEDURE data_node_maintenance(job_id int, config jsonb)
AS $$
DECLARE
    fs_oid oid;
BEGIN
    
    FOR fs_oid IN 
        SELECT fs.oid
        FROM pg_foreign_server fs, pg_foreign_data_wrapper fdw
        WHERE fs.srvfdw = fdw.oid
        AND fdw.fdwname = 'timescaledb_fdw';
    LOOP 
        RAISE NOTICE 'Healing foreign server with OID: %s', fs_oid;
        PERFORM _timescaledb_internal.remote_txn_heal_data_node(fs_oid);
        COMMIT;
    END LOOP;
END;
$$ LANGUAGE PLPGSQL;
```
would solve the problem.

Reply:
@davidkohn88 Thanks for the suggestion. I think the issue is that you want to commit after healing each data node, to ensure the state of the node is recorded locally on the AN and not later rolled back due to a failure on another data node. 

Still, not 100% sure it is an issue to begin with.

Reply:
You can stick a commit in that loop, but I believe in this case, it is going to commit after each perform, there's no implicit txn because we're no longer in a single statement.

Reply:
the way this is should be fine. I prevented the txn block was because 
```
BEGIN;
SELECT remote_txn_heal_data_node();
ROLLBACK;
```
is nonesense since the `commit transaction` 2pc command that's sometimes executed on the DN as part of this cannot be rolled back.

Reply:
Ok, so it sounds like the maintenance job in the PR is OK.